auth_supplier/views/auth_supplier_view.xml

```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <template id="auth_supplier.fields"
+              inherit_id="auth_signup.fields"
+              name="Auth Signup/ResetPassword form fields">
+        <xpath expr="//div[@class='form-group field-login']" position="before">
+            <div class="form-group field-type">
+                <label for="account_type" class="control-label">Account Type</label>
+                <select class="form-control" name="account_type">
+                    <option value="customer" selected="selected">Customer</option>
```

Comment:
You are right. It was just to point out the problem.
In fact, a module `auth_signup_account_type` could be the proper solution, where `ir.config.parameter` entries like `auth_signup_account_type.{TYPE}` to enable them on the UI, and each `TYPE` could be overloaded to load related data => `customer` binded to `{'customer': True}`, etc, and everyone should be able to implement the type they want in addition to the `default` one which is just there for the default `auth_signup` behaviour.

A new module for later!